modules/xmpp/JingleSessionPC.js

```diff
@@ -2538,7 +2558,16 @@ export default class JingleSessionPC extends JingleSession {
             logger.info(`${this} Sending source-add`);
             logger.debug(add.tree());
             this.connection.sendIQ(
-                add, null, this.newJingleErrorHandler(add), IQ_TIMEOUT);
+                add,
+                () => {
+                    this.room.eventEmitter.emit(XMPPEvents.SOURCE_ADD, this, ctx);
+                },
+                () => {
+                    this.newJingleErrorHandler(add);
```

Comment:
newJingleErrorHandler creates a method which needs to be called with an error

Reply:
Fixed.